pkg/resources/resources.go

```diff
@@ -1744,12 +1744,20 @@ func GetClusterNodeCIDRMaskSizeIPv6(cluster *kubermaticv1.Cluster) int32 {
 // GetNodePortsAllowedIPRanges returns effective CIDR range to be used for NodePort services for the given cluster
 // and provided allowed IP ranges coming from provider-specific API.
 func GetNodePortsAllowedIPRanges(cluster *kubermaticv1.Cluster, allowedIPRanges *kubermaticv1.NetworkRanges, allowedIPRange string) (res kubermaticv1.NetworkRanges) {
+	var ipRangesMap []string
 	if allowedIPRanges != nil {
-		res.CIDRBlocks = append(res.CIDRBlocks, allowedIPRanges.CIDRBlocks...)
+		ipRangesMap = allowedIPRanges.CIDRBlocks
 	}
+
 	if allowedIPRange != "" {
-		res.CIDRBlocks = append(res.CIDRBlocks, allowedIPRange)
+		for _, cidr := range allowedIPRanges.CIDRBlocks {
+			if cidr != allowedIPRange {
+				ipRangesMap = append(ipRangesMap, allowedIPRange)
```

Comment:
this actually won't work as expected, as it may append the same allowedIPRange multiple times (if `allowedIPRanges` contains more elements)